lib/mongo/crypt/binding.rb

```diff
@@ -1089,6 +1098,26 @@ def self.check_ctx_status(context)
           status.raise_crypt_error
         end
       end
+
+      # Checks that the specified data is a BSON::Document before serializing
+      # it to BSON to prevent errors from libmongocrypt
+      #
+      # @param [ Object ] data The data to be passed to libmongocrypt
+      #
+      # @raise [ Mongo::Error::CryptError ] If the data is not a BSON::Document
+      def self.validate_document(data)
+        return if data.is_a?(BSON::Document)
```

Comment:
Although I realize I said "BSON::Document" in the ticket, I believe Hash here will work also (with slightly different behavior in the driver implementation but both approaches are valid).